File: DiffProg_MyBook/julia_lab/Basic/test_class_r0.java
class Point {
    int x, y;
    public Point(int new_x, int new_y) {
        x = new_x;
        y = new_y;
    } 
    public String toString() {
        return getClass().getSimpleName() + "(" + x + ", " + y + ")";
    }
    public Point add(Point p) {
        return new Point(x + p.x, y + p.y);
    }
}

class test_class {
    public static void test1() {
        Point p1 = new Point(1, 2);
        System.out.println(p1);
    }
    public static void test2() {
        Point p1 = new Point(1, 2);
        Point p2 = new Point(3, 4);
        System.out.println(p1.add(p2)); // java doesn't support operator overloading!
        System.out.println(p1);
    }    
    public static void main(String[] args) {
        test1();
        System.out.println("--------------------");
        test2();
    }
}

File: DiffProg_MyBook/julia_lab/Basic/test_struct_r1/repository/test_class_r0.cpp
/*
[2022-2-2]
Class with public member variables
*/
#include <iostream>

class Point {
    public: 
        int x;
        int y;
};

void test1() {
    class Point p1; //= Point(1, 2);
    p1.x = 1;
    p1.y = 2;
    
    std::cout << "(" << p1.x << "," << p1.y << ")" << std::endl;
}

int main() {
    test1();
    return 0;
}

